lib/parser/objects.ts

```diff
@@ -1026,18 +1034,27 @@ export class OConfiguration extends ObjectBase implements IHasLibraries {
   libraries: OLibrary[] = [];
 }
 export function* scope(startObject: ObjectBase) {
+  for (const iterator of scopeWithVisibility(startObject)) {
+    yield iterator.obj;
+  }
+}
+export function* scopeWithVisibility(startObject: ObjectBase) {
```

Comment:
What is happening here? I am confused? Can you add a comment?

The one gives visibility information, the other strips that info away?
Can you add comment at two both functions at least?
And Can we keep it as a single iterator and yield [obj, directlyVisible] so that you can easily desctrucure it? (as array not object?)

Reply:
I took the liberty to cleanup a bit. Feel free to close resolve and merge If you think it is fine.